tests/test_spectrum_assignment.py

```diff
@@ -0,0 +1,378 @@
+    ind_max = len(oms_list[10].spectrum_bitmap.bitmap) - 1
+    nvalmin = oms_list[10].spectrum_bitmap.getn(0)
+    nvalmax = oms_list[10].spectrum_bitmap.getn(ind_max)
+    print(f'expected: {min(nmin, nvalmin)}, {max(nmax, nvalmax)}')
+    print(f'expected: {nmin, nmax}')
+    print(f'obtained after alignment: {nvalmin}, {nvalmax}')
+    assert nvalmin <= nmin and nvalmax >= nmax
+
+@pytest.mark.parametrize('nval1', [0, 15, 24])
+@pytest.mark.parametrize('nval2', [8, 12])
+def test_assign_and_sum(nval1, nval2, setup):
+    """ checks that bitmap sum gives correct result
+    """
+    network, oms_list = setup
+    grid = 0.00625e12
+    guardband = grid
+    mval = 4 # slot in 12.5GHz
+    freq_min = 193.1e12
+    freq_max = 193.1e12 + 24 * 0.00625e12
+    # arbitrary test on oms #10 and #11
+    # first reduce the grid to 24 center frequencies to ease reading when test fails
+    oms1 = oms_list[10]
+    oms1.update_spectrum(freq_min, freq_max, grid=grid, guardband=guardband)
+    oms2 = oms_list[11]
+    oms2.update_spectrum(freq_min, freq_max, grid=grid, guardband=guardband)
+    print('initial spectrum')
+    print(nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_min) * 1e-12,
+          nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_max) * 1e-12)
+    # checks initial values consistancy
+    ind_max = len(oms_list[10].spectrum_bitmap.bitmap) - 1
+    print('with guardband', oms_list[10].spectrum_bitmap.getn(0),
+          oms_list[10].spectrum_bitmap.getn(ind_max))
+    print('without guardband', oms_list[10].spectrum_bitmap.freq_index_min,
+          oms_list[10].spectrum_bitmap.freq_index_max)
+    test1 = oms1.assign_spectrum(nval1, mval)
+    print(oms1.spectrum_bitmap.bitmap)
+    # if requested slots exceed grid spectrum should not be assigned and assignment
+    # should return False
+    if ((nval1 - mval) < oms1.spectrum_bitmap.getn(0) or
+            (nval1 + mval-1) > oms1.spectrum_bitmap.getn(ind_max)):
+        print('assignment on part of bitmap is not allowed')
+        assert not test1
+        for elem in oms1.spectrum_bitmap.bitmap:
+            assert elem == 1
+    else:
+        oms2.assign_spectrum(nval2, mval)
+        print(oms2.spectrum_bitmap.bitmap)
+        test2 = bitmap_sum(oms1.spectrum_bitmap.bitmap, oms2.spectrum_bitmap.bitmap)
+        print(test2)
+        range1 = range(oms1.spectrum_bitmap.geti(nval1) - mval,
+                       oms1.spectrum_bitmap.geti(nval1) + mval -1)
+        range2 = range(oms2.spectrum_bitmap.geti(nval2) - mval,
+                       oms2.spectrum_bitmap.geti(nval2) + mval -1)
+        for elem in range1:
+            print(f'value should be zero at index {elem}')
+            assert test2[elem] == 0
+
+        for elem in range2:
+            print(f'value should be zero at index {elem}')
+            assert test2[elem] == 0
+
+def test_values(setup):
+    """ checks that oms.assign_spectrum(13,7) is (193137500000000.0, 193225000000000.0)
```

Comment:
This looks like an excellent use case for a docstring. Here's [an example how it's used in `gnpy.core.utils`](https://github.com/Telecominfraproject/oopt-gnpy/blob/develop/gnpy/core/utils.py#L187); these comments are auto-extracted by the test framework and run as a part of the test suite. No need to write the test in `tests/`, just add two lines to the method's docstring.

Reply:
Thanks for the suggestion I added them + some more